### PR TITLE
Sema: missing switch case fixits should print payload names explicitly instead of printing underscores. rdar://32121806

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: lib/Sema/TypeCheckSwitchStmt.cpp [hunk 1]
```diff
@@ -77,6 +77,10 @@ namespace {
       SpaceKind Kind;
       llvm::PointerIntPair<Type, 1, bool> TypeAndVal;
       Identifier Head;
+
+      // This is for type space only to help us print meaningful name, e.g.,
+      // tuple element name in fixits.
+      Identifier NameForPrinting;
```
Comment (CodaFi) [inline lib/Sema/TypeCheckSwitchStmt.cpp]: You probably could reuse the `Head` for this.

Comment (nkcsgexi) [inline lib/Sema/TypeCheckSwitchStmt.cpp]: So `Head` is not used at all for type space?

Comment (CodaFi) [inline lib/Sema/TypeCheckSwitchStmt.cpp]: Nope.  I assumed nobody would need it, so it prints as an underscore.

Comment (nkcsgexi) [inline lib/Sema/TypeCheckSwitchStmt.cpp]: Sounds good, will change.

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Review by CodaFi — APPROVED** [review]

**Pull request merged**